alf/algorithms/algorithm.py

```diff
@@ -297,11 +312,24 @@ def get_optimizer_info(self):
         optimizer_info = []
         for alg_name, opt, module_set in optimizer_and_module_sets:
             optimizer_info.append(
-                'Algorithm: "%s" Optimizer: "%s" Modules: "%s"' \
-                    % (alg_name,
-                       opt.get_config() if opt is not None else None,
-                       '; '.join(sorted([m.name for m in module_set if m is not None]))))
-        return '\n\n'.join(optimizer_info)
+                dict(
+                    Algorithm=alg_name,
+                    Optimizer=opt.get_config() if opt is not None else None,
+                    Modules=[
+                        dict(
+                            ModuleName=m.name,
+                            TrainableVariables=[
+                                v.name for v in _get_trainable_vars(m)
+                            ]) for m in module_set if m is not None
+                    ]))
+        json_pretty_str_info = json.dumps(
+            obj=optimizer_info, indent=2, default=get_json_type)
+
+        def _markdownify(paragraph):
+            return "    ".join(
+                (os.linesep + paragraph).splitlines(keepends=True))
+
+        return _markdownify(json_pretty_str_info)
```

Comment:
why _markdownify is needed?

Reply:
with _markdownify, tensorboard text summary will treat text content with  html `pre` label  and keep original format
<img height=256 src='https://user-images.githubusercontent.com/1267359/71857604-138f5b80-3123-11ea-90a1-e26bb8a675e4.png'>


without _markdownify

<img src='https://user-images.githubusercontent.com/1267359/71857504-c4492b00-3122-11ea-9d28-ad926df7ad3b.png'>